### PR TITLE
feat: sort logs in ascending order

### LINKED ISSUE #2787: Sort logs in ascending order
It helps in seeing the chronological order of logs when debugging

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #2787 

Demo:
https://www.loom.com/share/30da8e470bfe4684a938ec654a49b8c3?sid=e4101c9d-cb7d-4fac-98ad-ffe07eca12b3

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

**Comment by nityanandagohain** [comment]
Also please disable the `go to latest` button when ascending order is selected

**Comment by palashgdev** [comment]
> The sorting preferences should also be stored locally, it resets on refresh

for this one please update the url and initialise the state from the url if present other fallback to desc

**Comment by gitstart** [comment]
> In asc mode
> 
> * Next button -> instead of idLt , idGt should be used
> * Prev button -> instead of itGt, idLt should be used
> 

Hi @nityanandagohain, can you please elaborate on this?

**Comment by nityanandagohain** [comment]
@gitstart , in ASC mode the API params will change i.e

When you click on the next button the API request should send `idGt` in the payload and not `idLt`
And when prev button is clicked , API request should send `idLt` in the payload and not `idGt`

**Comment by gitstart** [comment]
Hi @nityanandagohain! PR has been updated with the requested changes, please re-review 🙏

**Review by nityanandagohain — COMMENTED** [review]
The change looks good to me

but, lets say in asc mode I am in nth page by clicking the `prev/next` button n number of times. Now If I try to click on the refresh button near the time button, it still renders the same page and there is not really an option for me to reset unless I manully refresh the browser page. But it should be fine for this use case.

Let me know what you guys think @palashgdev @ankitnayan

**Review by nityanandagohain — DISMISSED** [review]

**Comment by nityanandagohain** [comment]
Fixes https://github.com/SigNoz/signoz/issues/2176

File: frontend/src/container/LogDetailedView/ActionItem.tsx [hunk 1]
```diff
@@ -68,15 +69,27 @@ function ActionItem({
 		});
 
 		if (liveTail === 'STOPPED') {
+			const idConditions: Record<string, string> = {};
```
Comment (palashgdev) [inline frontend/src/container/LogDetailedView/ActionItem.tsx:72]: we have the same logic please avoid duplicates

Comment (palashgdev) [reply frontend/src/container/LogDetailedView/ActionItem.tsx:72]: reference: frontend/src/container/LogsSearchFilter/index.tsx

File: frontend/src/container/LogsSearchFilter/useSearchParser.ts [hunk 2]
```diff
@@ -39,7 +40,7 @@ export function useSearchParser(): {
 		(updatedQueryString: string) => {
 			history.replace({
 				pathname: history.location.pathname,
-				search: `?q=${updatedQueryString}`,
+				search: `?q=${updatedQueryString}&order=${order}`,
```
Comment (palashgdev) [inline frontend/src/container/LogsSearchFilter/useSearchParser.ts:43]: can we power the order and query from here

```
https://github.com/SigNoz/signoz/blob/1ded475b371f4d0a01874e3c195980f242a6ba4d/frontend/src/constants/query.ts
```

File: frontend/src/pages/Logs/index.tsx [hunk 3]
```diff
@@ -67,6 +71,16 @@ function Logs(): JSX.Element {
 		[handleViewModeOptionChange],
 	);
 
+	const handleChangeOrder = (value: 'asc' | 'desc'): void => {
+		dispatch({
+			type: SET_LOGS_ORDER,
+			payload: value,
+		});
+		const params = new URLSearchParams(window.location.search);
```
Comment (palashgdev) [inline frontend/src/pages/Logs/index.tsx:79]: don't use window use utils from react-router

File: frontend/src/pages/Logs/index.tsx [hunk 4]
```diff
@@ -101,6 +115,15 @@ function Logs(): JSX.Element {
 										<Button>Format</Button>
 									</Popover>
 								)}
+
+								<Select
+									style={defaultSelectStyle}
+									defaultValue={order}
+									onChange={handleChangeOrder}
+								>
+									<Select.Option key="desc">Descending</Select.Option>
```
Comment (palashgdev) [inline frontend/src/pages/Logs/index.tsx:124]: can we have enum please just to avoid magic strings

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Comment by gitstart** [comment]
Hi @palashgdev, we have made the requested change. Please review

**Review by palashgdev — APPROVED** [review]

**Review by palashgdev — APPROVED** [review]

**Pull request merged**